source/change-streams/change-streams.rst

```diff
@@ -547,6 +611,14 @@ In the above example, not sending ``startAtOperationTime`` will result in the ch
 the changes that occurred while the server and client are partitioned. By sending ``startAtOperationTime``,
 the server will know to include changes from that previous point in time.
 
+---------------------------------------------------------------
+Why do we provide a public getter for the current resume token?
+---------------------------------------------------------------
+
+Users should be able to implement their own custom resume logic (e.g. save the resume token in a file and restart later). `ChangeStream::resumeToken`_ explains how we set the current ``resumeToken`` to use on resume. When possible, resume tokens are taken from the ``resumeTokens`` of an aggregate/getMore response instead of the ``_id`` of a ``ChangeDocument``. The resume tokens in ``resumeTokens`` may point to more recent entries in the oplog. This makes it much less likely that a change stream cursor will roll off of the oplog and helps performance of resumes.
+
+The ``resumeTokens`` array may not be conveniently accessible to clients (outside of using APM), so the public getter provides an easy way to implement custom resume logic.
+
```

Comment:
TODO: Add FAQ entry explaining how `minPromisedResumeToken` appears in the `resumeTokens` array for both empty and non-empty batches. In particular, it will replace the resume token of the final document for non-empty batches.

See https://github.com/mongodb/specifications/pull/413#discussion_r233237081 for more context.

Reply:
Added FAQ entry explaining why we need to expose postBatchResumeToken.